src/test/unit/profile-tree.test.js

```diff
@@ -526,6 +533,82 @@ describe('inverted call tree', function() {
   });
 });
 
+describe('diffing trees', function() {
+  function getProfile() {
+    const profile = getMergedProfileFromTextSamples(
+      `
+      A  A  A
+      B  B  C
+      D  E  F
+    `,
+      `
+      A  A  A
+      B  B  B
+      G  I  E
+    `
+    );
+    return profile;
+  }
+
+  it('displays a proper call tree, including nodes with totalTime = 0', () => {
+    const profile = getProfile();
+    const callTree = callTreeFromProfile(profile, /* threadIndex */ 2);
+    const formattedTree = formatTree(callTree);
+    expect(formattedTree).toEqual([
+      '- A (total: —, self: —)',
+      '  - B (total: 1, self: —)',
+      '    - D (total: -1, self: -1)',
+      '    - G (total: 1, self: 1)',
+      '    - I (total: 1, self: 1)',
+      '  - C (total: -1, self: —)',
+      '    - F (total: -1, self: -1)',
+    ]);
+
+    // There's no A -> B -> E node because the diff makes it completely disappear.
+    expect(formattedTree).not.toContainEqual(expect.stringMatching(/^\s*- E/));
```

Comment:
I checked that this check was failing for a present node :)

Reply:
Thanks for the test here.